src/main/java/org/jabref/logic/cleanup/RenamePdfCleanup.java

```diff
@@ -121,13 +121,14 @@ public RenamePdfCleanup(boolean onlyRelativePaths, BibDatabaseContext databaseCo
             if (renameSuccessful) {
                 changed = true;
 
-                //Change the path for this entry
+                // Change the path for this entry
                 String description = oldLinkedFile.getDescription();
                 String type = oldLinkedFile.getFileType();
 
-                //We use the file directory (if none is set - then bib file) to create relative file links
-                Optional<Path> settingsDir = databaseContext.getFirstExistingFileDir(fileDirectoryPreferences);
-                settingsDir.ifPresent(path -> newFileList.add(new LinkedFile(description, path.relativize(newPath).toString(), type)));
+                // We use the file directory (if none is set - then bib file) to create relative file links.
+                // The .get() is legal without check because the method will always return a value.
+                Path settingsDir = databaseContext.getFirstExistingFileDir(fileDirectoryPreferences).get();
```

Comment:
Could you add a test case for that?
Just to make sure we don't miss anything

Reply:
What do you mean exactly? The tests already cover this to some extend. https://codecov.io/gh/JabRef/jabref/compare/c416818deba486605779a375a8f09f605c8dec9c...e4604444bdf8f1ed93678736183248d25924ae93/src/src/main/java/org/jabref/logic/cleanup/RenamePdfCleanup.java

Reply:
Sorry, did not see that! I merge it in now